### PR TITLE
feat: remark-lint rule converter

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task]()

---

## Description

Add a method that converts `remark-lint` rules into valid content conformance rules.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203331966445547

### CODE REVIEW

File: packages/content-conformance/src/rules.ts [hunk 1]
```diff
@@ -130,3 +131,40 @@ export async function loadRule(
...
+
+  const convertedRule: LoadedConformanceRule = {
+    level,
+    type: 'content',
+    id: rule.name,
+    description: '',
+    executor: {
+      async contentFile(file) {
+        /**
+         * The rule function accepts a tree, a file, and a callback. By convention remark-lint rules already handle attaching messages, so there's nothing else we need to do here to handle proper reporting.
```
Comment (thiskevinwang) [inline packages/content-conformance/src/rules.ts:162]: 🧠 ⭐

File: packages/content-conformance/src/__tests__/rules.test.ts [hunk 2]
```diff
@@ -86,3 +89,35 @@ describe('loadRules', () => {
...
+        "executor": {
+          "contentFile": [Function],
+        },
+        "id": "remark-lint:no-h1",
+        "level": "error",
+        "type": "content",
+      }
+    `)
+  })
+})
```
Comment (thiskevinwang) [inline packages/content-conformance/src/__tests__/rules.test.ts:123]: 🙌

File: packages/content-conformance/src/rules.ts [hunk 3]
```diff
@@ -40,35 +40,52 @@ export async function loadRules(
...
+      if (error.message.includes('Cannot find module')) {
+        console.error(
+          `[content-conformance] error loading remark-lint rule: ${ruleNameOrPath}. Is it installed?
+
+  npm install --save-dev ${ruleNameOrPath}`
+        )
+      }
+      return
+    }
+  }
```
Comment (thiskevinwang) [inline packages/content-conformance/src/rules.ts:87]: Do we need an `else` or early-return check here? 
- If we successfully load a `remark-lint-` rule, we should not try to load the internal rule

File: packages/content-conformance/src/rules.ts [hunk 4]
```diff
@@ -128,6 +145,9 @@ export async function loadRule(
     // local rule - relative not found
   }
 
+  // ensure the rule severity level gets included so we can reference it when reporting
+  rule.level = level
```
Comment (thiskevinwang) [inline packages/content-conformance/src/rules.ts:149]: This `level` assignment looks like it is only needed to patch the `internal rule` case right?
- `convertRemarkLintRule()` returns the level properly.

NIT: Maybe move this alongside the internalRule assignment

```diff
  rule = internalRule
+ rule.level = level
```

File: packages/content-conformance/src/rules.ts (hunk 3 above)
Comment (BRKalow) [reply packages/content-conformance/src/rules.ts:87]: Yeah, good call!

File: packages/content-conformance/src/rules.ts (hunk 4 above)
Comment (BRKalow) [reply packages/content-conformance/src/rules.ts:149]: There are other cases that need this assignment: internal rules and local rules. They are loaded below the remark lint rule `try/catch` block. I can likely refactor this to make it more clear though 👀

File: packages/content-conformance/src/rules.ts (hunk 3 above)
Comment (BRKalow) [reply packages/content-conformance/src/rules.ts:87]: I've added an early return if the rule loads successfully.

**Comment by BRKalow** [comment]
Going to merge this, feel free to leave comments and I'll address in a follow-up.

**Pull request merged**